src/Microsoft.AspNet.Http.Core/Extensions/MapWhenMiddleware.cs

```diff
@@ -20,27 +20,13 @@ public MapWhenMiddleware([NotNull] RequestDelegate next, [NotNull] MapWhenOption
 
         public async Task Invoke([NotNull] HttpContext context)
         {
-            if (_options.Predicate != null)
+            if (_options.Predicate(context))
```

Comment:
This doesn't need a null check anymore? The setter for this property doesn't seem to guard against null.